tests/framework/extensions/prime/primechecks.go

```diff
@@ -0,0 +1,76 @@
+package primechecks
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	client "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	rancherBrand         = "suse"
+	PodResourceSteveType = "pod"
+)
+
+// CheckUIBrand checks the UI brand of Rancher Prime. If the Rancher instance is not Rancher Prime, the UI brand should be blank.
+func CheckUIBrand(client *rancher.Client, isPrime bool, brand *client.Setting) {
+	if isPrime {
+		if rancherBrand == brand.Value {
+			logrus.Infof("Rancher Prime UI brand: %s", rancherBrand)
+		}
+	} else {
+		logrus.Infof("Rancher UI brand is blank")
+	}
+}
+
+// CheckVersion checks the if Rancher Prime is set to true and the version of Rancher.
+func CheckVersion(isPrime bool, rancherVersion string, serverConfig *rancherversion.Config) {
+	if isPrime {
+		if isPrime == serverConfig.IsPrime {
+			logrus.Infof("Rancher Prime: %t | Version: %s", isPrime, rancherVersion)
+		}
+	} else {
+		logrus.Infof("Rancher Prime: %t | Version: %s", isPrime, rancherVersion)
+	}
+}
+
+// CheckSystemDefaultRegistry checks if the system default registry is set to the expected value.
+func CheckSystemDefaultRegistry(isPrime bool, primeRegistry string, registry *client.Setting) {
+	if isPrime {
+		if primeRegistry == registry.Value {
+			logrus.Infof("Rancher Prime system default registry: %s", primeRegistry)
+		}
+	} else {
+		logrus.Infof("Rancher system default registry is blank")
+	}
+}
+
+// CheckLocalClusterRancherImages checks if the Rancher images are set to the expected registry.
+func CheckLocalClusterRancherImages(client *rancher.Client, isPrime bool, rancherVersion, clusterID string) error {
```

Comment:
I can update the name if it helps. The function is just looking to see if the defined registry in the local cluster is being used for workloads in the system project page. It's the same logic we use for checking all pods state, just getting the image instead.

Reply:
I think, if we apply the same logic here as I am asking for in, say `CheckVersion` then we should be good. But to be clear, is it ok if some images are not using the specified registry? i.e. if you assert that all system-level images are using the registry, is that a fair assertion to make? or are there exceptions?

Reply:
You're correct, tbh, I think that's a separate issue that the test highlighted that wasn't caught before in manual testing. Because I would expect the same as well, this test simply just grabs the image and it did on my fresh install.

Reply:
I agree, this doesn't verify, it logs, maybe returns like the above, this can have something that can be validated in the tests with assert

Reply:
Resolved.